src/Infrastructure/InternetDomainNameValidator.php

```diff
@@ -12,8 +12,17 @@
  */
 class InternetDomainNameValidator implements DomainNameValidator {
 
+	private const BANNED_DOMAINS = [
+		'example.com'
+	];
+
 	public function isValid( string $domain ): bool {
-		return checkdnsrr( $domain, 'MX' ) || checkdnsrr( $domain, 'A' ) || checkdnsrr( $domain, 'SOA' );
+		return !$this->isDomainBanned( $domain ) &&
+			( checkdnsrr( $domain, 'MX' ) || checkdnsrr( $domain, 'A' ) || checkdnsrr( $domain, 'SOA' ) );
```

Comment:
For consistencies sake you could also put this check into a `domainHasDnsEntry` method, so the `isValid` reads more nicely.

Reply:
Yeah, that makes the code much nicer for sure, I have made that change!